### PR TITLE
Projects API enhancements

### PR DESCRIPTION
**Pull request by Nitish145**
#### Describe the changes you have made in this PR -
1. Serialize `author_name` in projects serializer.
2. Add selective including of author details.
3. Update/Add specs for above mentioned changes.

Note: Please check **Allow edits from maintainers.** if you would like us to assist in the PR.

### CODE REVIEW

**Comment by Nitish145** [comment]
@tachyons please have a look!!

File: app/controllers/api/v1/projects_controller.rb [hunk 1]
```diff
@@ -14,6 +14,7 @@ class Api::V1::ProjectsController < Api::V1::BaseController
   before_action :sort, only: %i[index user_projects featured_circuits]
 
   SORTABLE_FIELDS = %i[view created_at].freeze
+  ALLOWED_TO_BE_INCLUDED = %i[author].freeze
```
Comment (tachyons) [inline app/controllers/api/v1/projects_controller.rb:17]: What about WHITELISTED_INCLUDE_ATTRIBUTES ?

Comment (Nitish145) [reply app/controllers/api/v1/projects_controller.rb:17]: Will make the change.. @tachyons

Comment (Nitish145) [reply app/controllers/api/v1/projects_controller.rb]: Done! @tachyons

**Pull request merged**